front/lib/resources/data_source_resource.ts

```diff
@@ -376,6 +376,9 @@ export class DataSourceResource extends ResourceWithVault<DataSource> {
   }
 
   canWrite(auth: Authenticator) {
+    if (this.vault.isRegular()) {
+      return auth.canWrite([this.vault.acl()]);
+    }
     return auth.isBuilder() && auth.canWrite([this.vault.acl()]);
```

Comment:
I presume this can be replaced by 👇 so permissions are not spread across the codebase.
```suggestion
    return this.vault.canWrite(auth);
```

Reply:
Nope for some reason as user I have the write permission on global. 
I agree this looks cleaner and we expect `vault.canWrite(auth);` to be false for Global if not builder.

Reply:
yes would be better to handle all cases in vault.canWrite directly . All members -> read/write except for global, everybody->read and builder->write

Reply:
Actually yes def since this was added to the vault one!